sycl/include/CL/sycl/backend/level_zero.hpp

```diff
@@ -35,6 +35,15 @@ template <> struct interop<backend::level_zero, program> {
   using type = ze_module_handle_t;
 };
 
+template <bundle_state State>
+struct interop<backend::level_zero, kernel_bundle<State>> {
```

Comment:
Why are you adding these "interop"? What is using them? The "interop" should go away in favor of "input_t", etc

Reply:
A lot of code in this repository is still using them. Besides, it is a user-visible structure, so client code may also use it (e.g. when code style advises against excessive `auto` usage). I can replace all those usages with `backend_traits` if you'd like to, but in a separate patch. Besides, we'll still need some sort of structure for both input and return types, since one can not specialize `using` constructs.

Reply:
I meant to ask why add new "interop" (being deprecated) for the "kernel_bundle" specifically. We should add new things with "backend_traits" already, and replace old ones eventually (I agree in a separate patch, and probably only at major compiler update).

Reply:
@smaslov-intel as I said before, one cannot specialize the `using` clause, and I wanted to avoid code duplication. Adding new specialization for `interop` was the easiest way.

Reply:
I still don't understand what breaks if you just remove the lines 38-41?

Reply:
This is how `backend_traits` are declared now:
https://github.com/intel/llvm/blob/be7c1cbcdd6f59ac2d3d56d6aa5c72c19d0d8cb7/sycl/include/CL/sycl/backend.hpp#L28-L32

Ideally, we'd have separate structures for both input and output types, for each backend. But at the time we don't have support for the types, that are different for input and output, so I decided to re-use existing structure. And `ze_module_handle_t` seem to perfectly match both input and output for `kernel_bundle`.